src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewModel.swift

```diff
@@ -0,0 +1,111 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import UIKit
+import Combine
+
+class ExposureSubmissionTestResultConsentViewModel {
+	// MARK: - Init
+	
+	init(
+		supportedCountries: [Country],
+		exposureSubmissionService: ExposureSubmissionService
+	) {
+		
+		self.exposureSubmissionService = exposureSubmissionService
+		
+		self.supportedCountries = supportedCountries.sorted { $0.localizedName.localizedCompare($1.localizedName) == .orderedAscending }
+		
+		self.exposureSubmissionService.isSubmissionConsentGivenPublisher.sink { isSubmissionConsentGiven in
+			self.consentSwitch.isOn = isSubmissionConsentGiven
+		}.store(in: &cancellables)
+	}
+
+
+	// MARK: - Properties
+	
+	@objc
+	func consentStateChanged(switchState: UISwitch) {
+		exposureSubmissionService.setSubmissionConsentGiven(consentGiven: switchState.isOn)
+	}
+		
+	// MARK: - Private
+	
+	private var cancellables: Set<AnyCancellable> = []
+		
+	private let supportedCountries: [Country]
+	
+	private var exposureSubmissionService: ExposureSubmissionService
+	
+	private let consentSwitch = UISwitch()
+	
+	// MARK: - Internal
+	
+	var dynamicTableViewModel: DynamicTableViewModel {
+		DynamicTableViewModel.with {
+			$0.add(
+				.section(
+					separators: .none,
+					cells: [
+						.title2(
+							text: AppStrings.AutomaticSharingConsent.switchTitle,
+							color: nil,
+							accessibilityIdentifier: nil,
+							accessibilityTraits: .header,
+							configure: { _, cell, _ in
+								cell.accessoryView = self.consentSwitch
+								self.consentSwitch.onTintColor = .enaColor(for: .tint)
+								self.consentSwitch.addTarget(self, action: #selector(self.consentStateChanged), for: .valueChanged)
+							}
+						),
```

Comment:
capture self inside this closure might leak?

Reply:
Put the weak reference[weak]. Solved.